engine/lua/src/cs/dmsdk/lua/lual.cs

```diff
@@ -0,0 +1,98 @@
+// Copyright 2020-2024 The Defold Foundation
+// Copyright 2014-2020 King
+// Copyright 2009-2014 Ragnar Svensson, Christian Murray
+// Licensed under the Defold License version 1.0 (the "License"); you may not use
+// this file except in compliance with the License.
+//
+// You may obtain a copy of the License, together with FAQs at
+// https://www.defold.com/license
+//
+// Unless required by applicable law or agreed to in writing, software distributed
+// under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+// CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+
+using System.Runtime.InteropServices;
+using System.Reflection.Emit;
+
+namespace dmSDK.Lua {
+    public unsafe partial class LuaL
+    {
+        [StructLayout(LayoutKind.Sequential, CharSet = CharSet.Ansi, Pack=1)]
+        public unsafe struct Reg
+        {
+            public IntPtr   name;
+            public IntPtr   func;
+        }
+
+        // A way to help pass managed strings
+        public struct RegHelper
+        {
+            public String?  name;
+            public IntPtr   func;
+        }
+
+        public static void Register(Lua.State* L, String libname, RegHelper[] inregs)
+        {
+            Reg[] regs = new Reg[inregs.Length];
+            for (int i = 0; i < inregs.Length; i++)
+            {
+                if (inregs[i].name != null)
+                {
+                    regs[i].name = (IntPtr)Marshal.StringToHGlobalAnsi(inregs[i].name);
+                }
+                else
+                {
+                   regs[i].name = 0;
+                }
+                regs[i].func = inregs[i].func;
+            }
+
+            fixed (Reg* pregs = regs)
+            {
+                LuaL.register(L, libname, pregs);
+            }
+
+            for (int i = 0; i < regs.Length; i++)
+            {
+                if (regs[i].name != 0)
+                    Marshal.FreeHGlobal(regs[i].name);
+            }
+        }
```

Comment:
Again, due to the marshalling of strings, I created a helper function to help with the string manipulations.
So, it's recommended to use `LuaL.Register()` (which then calls the `LuaL.register()` internally.

Reply:
Maybe it worth adding it as a comment for this function :
```
    /// <summary>
    /// info here
    /// </summary>
    ```
    
    https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/xmldoc/